### ISSUE TITLE
4.11.0 minimal bundle => No renderer for attachment for screen reader of type XY

### ISSUE THREAD

**Issue by martinoss** [microsoft/BotFramework-WebChat#3625]
## Screenshots

![image](https://user-images.githubusercontent.com/2587942/101178469-51ee0d00-3649-11eb-9b54-dda25815304e.png)

![image](https://user-images.githubusercontent.com/2587942/101178872-ccb72800-3649-11eb-957f-56c464ef952e.png)

## Version

4.11.0 / NPM ('botframework-webchat/lib/index-minimal') / (but it also happening using CDN minimal bundle)

## Describe the bug

We're using the minimal bundle for our webchat. We're rendering all of the messages by our self.
Since we've updated to 4.11.0, all activities are still rendered as before, but there are errors in the console about missing renderers for screen readers, like:

`ErrorBox.js?ffa8:44 Error: No renderer for attachment for screen reader of type "application/vnd.microsoft.card.hero"`

## Steps to reproduce

1. Run this sample [01.getting-started/b.minimal-bundle](https://github.com/microsoft/BotFramework-WebChat/blob/master/samples/01.getting-started/b.minimal-bundle)
2. Open console in developer tools
3. Type "help"
4. Observe errors

## Expected behavior

I expect to have one of the following:
* renderers in that bundle 
* no renderers but no errors
* documented a way how to suppress/ignore errors from missing renderers => **any idea to work around the issue?**

[Bug]

**Comment by corinagum** [comment]
Hi @martinoss, thanks for filing this issue. I have a repro of this, and will investigate the problem. I don't think I'll get an answer to this thread today, but I will get it in as soon as possible early next week.

**Comment by corinagum** [comment]
`Composer.js`:
![image](https://user-images.githubusercontent.com/14900841/101418456-cbcd0300-38a2-11eb-8621-b8a90d08e759.png)

A couple of things:
1. I agree that these errors should be suppress-able, particularly for situations like using minimal bundle. It looks like you can disable the warning by adding the following middleware: (code from `__tests__/html/middleware.liveRegionAttachment.noWarning.returnFalse.html`)
![image](https://user-images.githubusercontent.com/14900841/101419262-7c87d200-38a4-11eb-88c8-c87083c3cffe.png)
 But making this necessary for minimal bundle seems a bit much.
   1. At the very least, we should have a sample or documentation for the above.
    1. In the end, if the customer doesn't want to make a live renderer for the attachment, it's not our job to force them. If we make it suppress-able, at least the customer is guaranteed to see these errors, and will have to take action to suppress them (though of course, we recommend making the renderer instead)
1. We should verify that the screen reader error in particular is only showing up for: 
   1. No renderer at all
   1. Using a middleware renderer, but doesn't have a screen reader one
1. Create a sample for screen reader renderer. If the customer is using their own renderer, there should be an easy way to verify how to write the screenreader middlware, particularly because a11y is a high priority for this project. 

In the past, we have asserted that we won't guarantee our **samples** being accessible as they are code examples only, and not meant to be used in production. I think this is a different situation in that we are 'enforcing' an accessible live region/screen reader renderer, and the customer needs to be able to easily find how to do that.

@martinoss please try the test code from the second screencap above. If this does not suppress the errors, feel free to report back here.

**Comment by martinoss** [comment]
@corinagum Thank your for investigating this. Your suggestion successfully suppress the errors, many thanks for this.
I also want to let you know that I had to implement the same workaround for activity type "messageReaction" in the activity middleware.

```js
activityMiddleware = () => (next: any) => (...setupArgs) => {
  const [card] = setupArgs
  
  if (card.activity.type === "messageReaction") {
    
    // There is no renderer for this activity, so skip it to avoid console errors.
    return false;
  }

  return next(...setupArgs)      
};
```

**Comment by corinagum** [comment]
Thanks for reporting back. 👍 

> I had to implement the same workaround for activity type "messageReaction" in the activity middleware.

Web Chat doesn't natively support `messageReaction`, so this is the expected behavior when receiving that activity type.

**Comment by corinagum** [comment]
I added `needs-team-attention` label to discuss how to appropriately handle lack of screen reader renderer in minimal bundle. See first item in https://github.com/microsoft/BotFramework-WebChat/issues/3625#issuecomment-740261281 for potential R12 planning.

**Comment by corinagum** [comment]
1. change to warning, not error
2. suppress warning? React props.ref - console log ==> warning, **align with this behavior** / production
   1. npm build - production vs start (dev)
   2. if suppressable, write sample
   3. default props (not style options)
   4. suppress red box in production & no announcement in screenreader (when WC dep in CRA)
3. Sample for suppressing warnings (from test code)
 5. dev mode, non-suppressed - have screenreader renderer saying activity could not be rendered
